sn_node/src/put_validation.rs

```diff
@@ -89,6 +91,7 @@ impl Node {
     pub(crate) async fn validate_and_store_chunk(
         &self,
         chunk_with_payment: ChunkWithPayment,
+        validate_payment: bool,
```

Comment:
as mentioned at the discussion https://maidsafe.discourse.team/t/pricing-data-simply/808/13
For ReplicatedData, we may still need to validate the payment is valid, just the `amount` doesn't matter
For original `PutRecord`, we not only need to validate the payment is valid, but also need to confirm the amount is correct

Reply:
Updated for this 👍